GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTNanopbHelpers.m

```diff
@@ -188,6 +195,76 @@ gdt_cct_IosClientInfo GDTCCTConstructiOSClientInfo() {
+        if (networkConnectionInfo.mobile_subtype !=
+            gdt_cct_NetworkConnectionInfo_MobileSubtype_UNKNOWN_MOBILE_SUBTYPE) {
+          networkConnectionInfo.has_mobile_subtype = 1;
+        }
+      }
+    }
+  } else if (GDTCORPlatformIsMacOS() || GDTCORPlatformIsTvOS() || GDTCORPlatformIsWatchOS()) {
+    SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+    // Currently set all non-iOS platform network type to WIFI if network is reachable.
+    if (reachabilityFlags & kSCNetworkReachabilityFlagsReachable) {
+      networkConnectionInfo.has_network_type = 1;
+      networkConnectionInfo.network_type = gdt_cct_NetworkConnectionInfo_NetworkType_WIFI;
+    }
+  }
+  return networkConnectionInfo;
+}
+
+gdt_cct_NetworkConnectionInfo_NetworkType GDTCCTNetworkConnectonInfoNetworkType() {
+  gdt_cct_NetworkConnectionInfo_NetworkType networkType =
+      gdt_cct_NetworkConnectionInfo_NetworkType_NONE;
+  SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+  if (reachabilityFlags & kSCNetworkReachabilityFlagsIsWWAN) {
+    networkType = gdt_cct_NetworkConnectionInfo_NetworkType_MOBILE;
+  } else {
+    networkType = gdt_cct_NetworkConnectionInfo_NetworkType_WIFI;
+  }
+  return networkType;
+}
+
+gdt_cct_NetworkConnectionInfo_MobileSubtype GDTCCTNetworkConnectionInfoNetworkMobileSubtype() {
+  static NSDictionary<NSString *, NSNumber *> *ctRadioAccessTechnologyToNetworkSubType;
+  static dispatch_once_t onceToken = 0;
+  dispatch_once(&onceToken, ^{
+    ctRadioAccessTechnologyToNetworkSubType = @{
+      CTRadioAccessTechnologyGPRS : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_GPRS),
+      CTRadioAccessTechnologyEdge : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_EDGE),
+      CTRadioAccessTechnologyWCDMA :
+          @(gdt_cct_NetworkConnectionInfo_MobileSubtype_UNKNOWN_MOBILE_SUBTYPE),
+      CTRadioAccessTechnologyHSDPA : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_HSDPA),
+      CTRadioAccessTechnologyHSUPA : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_HSUPA),
+      CTRadioAccessTechnologyCDMA1x : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_CDMA),
+      CTRadioAccessTechnologyCDMAEVDORev0 : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_EVDO_0),
+      CTRadioAccessTechnologyCDMAEVDORevA : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_EVDO_A),
+      CTRadioAccessTechnologyCDMAEVDORevB : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_EVDO_B),
+      CTRadioAccessTechnologyeHRPD : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_EHRPD),
+      CTRadioAccessTechnologyLTE : @(gdt_cct_NetworkConnectionInfo_MobileSubtype_LTE),
+    };
+  });
+  NSString *networkInfoString = GDTCCTNetworkInfo().currentRadioAccessTechnology;
+  NSNumber *networkMobileSubtype = ctRadioAccessTechnologyToNetworkSubType[networkInfoString];
+  return networkMobileSubtype.intValue;
+}
+
+CTTelephonyNetworkInfo *GDTCCTNetworkInfo() {
```

Comment:
for macOS, tvOS, and watchOS you'll need to write a wrapper class (probably in GDTCORPlatform) around CTTelephonyNetworkInfo to return from this function.